deployment/keystone/state.go

```diff
@@ -19,6 +19,10 @@ import (
 type GetContractSetsRequest struct {
 	Chains      map[uint64]deployment.Chain
 	AddressBook deployment.AddressBook
+
+	// AddressFilterer is a function that filters the addresses a given address book.  Filtering
+	// mutates the input map so the input map should be a copy of the original map.
+	AddressFilterer func(map[string]deployment.TypeAndVersion)
```

Comment:
add a test for the case of multiple addresses of the same type and version and no filter

if i read the code correctly, that will cause the last one to be choosen (line#115)

instead we need to error if there are multiple instances of any contract.

that will be a backward incompatible change in the downstream repo: after you deploy an ocr3 contract all other invocations will be broken if they don't specify an input filter

that means we need to manage the filter configuration and plumb the filter func to all the existing config structs.

this seems error prone and burdensome if we end up will a handful of cor3 contracts

alternatively,
we can change the ContractSet to support a map[addr]*contract. and then all the call sites in this package that actually have use of ocr will have a config struct will non-optional addr/s

this seems better (though i originally opposed change the constructSet b/c i forseen this problem)